powermon/ports/testport.py

```diff
@@ -39,20 +40,21 @@ def disconnect(self) -> None:
         return
 
     def send_and_receive(self, command: Command) -> Result:
-        command_defn = command.command_definition
+        command_defn : CommandDefinition = command.command_definition
         
         result = Result(command.code)
 
-        if command_defn is not None:
+        if command_defn is not None and command_defn.test_responses is not None:
```

Comment:
maybe this should be in command object, e.g. command.get_test_response(self.response_number) style??

Reply:
This line of code made it in because I had an error parsing the configuration and my test responses were None causing errors. I will revert it since I made it so test responses need to be present or powermon won't run at all.

Reply:
Some protocols don't have test responses for all commands. I was more thinking of moving the logic to the command object